integration/src/main/java/ai/djl/integration/tests/ndarray/NDArrayShapesManipulationOpTest.java

```diff
@@ -77,26 +79,40 @@ public void testReshape() {
             NDArray array = manager.create(new float[] {1f, 2f, 3f, 4f, 5f, 6f});
             NDArray expected =
                     manager.create(new float[] {1f, 2f, 3f, 4f, 5f, 6f}, new Shape(2, 1, 1, 3));
-            Assert.assertEquals(array.reshape(2, 1, 1, 3), expected);
-            Assert.assertEquals(array.reshape(-1, 1, 1, 3), expected);
+            Assertions.assertAlmostEquals(array.reshape(2, 1, 1, 3), expected);
+            // only MXNet, PyTorch and TensorFlow support -1
+            if (TestUtils.isEngine("MXNet")
+                    && TestUtils.isEngine("PyTorch")
+                    && TestUtils.isEngine("TensorFlow")) {
+                Assertions.assertAlmostEquals(array.reshape(-1, 1, 1, 3), expected);
+            }
 
             // multi-dim
             array = manager.create(new float[] {1f, 2f, 3f, 4f, 5f, 6f}, new Shape(3, 2));
             expected = manager.create(new float[] {1f, 2f, 3f, 4f, 5f, 6f}, new Shape(2, 3));
-            Assert.assertEquals(array.reshape(new Shape(2, 3)), expected);
-            Assert.assertEquals(array.reshape(new Shape(2, -1)), expected);
+            Assertions.assertAlmostEquals(array.reshape(new Shape(2, 3)), expected);
+            // only MXNet, PyTorch and TensorFlow support -1
+            if (TestUtils.isEngine("MXNet")
+                    && TestUtils.isEngine("PyTorch")
+                    && TestUtils.isEngine("TensorFlow")) {
+                Assertions.assertAlmostEquals(array.reshape(new Shape(2, -1)), expected);
+            }
 
             // scalar
             array = manager.create(5f);
             expected = manager.create(new float[] {5f});
-            Assert.assertEquals(array.reshape(1), expected);
+            Assertions.assertAlmostEquals(array.reshape(1), expected);
             expected = manager.create(new float[] {5f}, new Shape(1, 1, 1));
-            Assert.assertEquals(array.reshape(1, -1, 1), expected);
-
+            // only MXNet, PyTorch and TensorFlow support -1
+            if (TestUtils.isEngine("MXNet")
+                    && TestUtils.isEngine("PyTorch")
+                    && TestUtils.isEngine("TensorFlow")) {
+                Assertions.assertAlmostEquals(array.reshape(1, -1, 1), expected);
```

Comment:
Should those engine throw UnsupportedException when passing -1?

Reply:
ok let's throw exception first and add TODO there